### PR TITLE
Remove All V2

### PR DESCRIPTION
**Pull request by alecholmez**
Signed-off-by: Alec Holmes <alecholmez@me.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,15 +58,12 @@ in the same environment as CI. This makes sure to produce a consistent set of ge
...
 
-For authoring changes, make changes to v2 and at the end, use `make create_version` to create the v3 specific files.
-Make sure to run `make build` and `make test` to identify and fix failures.
-
-When v2 version is frozen in the future, we will change the experience such that changes will need to happen to v3 and autogen will create the v2 version instead.
+### Deprecated
 
+`V2` control-plane code has been removed and will no longer be supported. For previous conversations on support for various xDS versions, see here: 
+- [here](https://docs.google.com/document/d/1ZkHpz6DwEUmAlG0kb2Mgu4iaeQC2Bbb0egMbECoNNKY/edit?ts=5e602993#heading=h.15nsmgmjaaml) 
+- [here](https://envoyproxy.slack.com/archives/C7LDJTM6Z/p1582925082005300),
```
Comment (snowp) [inline README.md:66]: Maybe include a note that in order to use V2, a previous SHA should be used

File: internal/example/server.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ func registerServer(grpcServer *grpc.Server, server serverv3.Server) {
 }
 
 // RunServer starts an xDS server at the given port.
-func RunServer(ctx context.Context, srv3 serverv3.Server, port uint) {
+func RunServer(ctx context.Context, srv3 server.Server, port uint) {
```
Comment (snowp) [inline internal/example/server.go:50]: srv3 seems to be a V3 remnant

**Review by snowp — CHANGES_REQUESTED** [review]
Thanks this looks good besides a few nits. Maybe do a search for V3 or just 3 to see if there are any more references to V3 that should be updated.

File: Makefile [hunk 3]
```diff
@@ -26,53 +26,33 @@ cover:
...
-
-integration.rest.v3: $(BINDIR)/test $(BINDIR)/upstream
 	env XDS=rest SUFFIX=v3 build/integration.sh
 
-integration.ads.tls: $(BINDIR)/test $(BINDIR)/upstream
-	env XDS=ads build/integration.sh -tls
-
-integration.xds.mux.v3: $(BINDIR)/test $(BINDIR)/upstream
+integration.xds.mux: $(BINDIR)/test $(BINDIR)/upstream
 	env XDS=xds SUFFIX=v3 build/integration.sh -mux
```
Comment (snowp) [inline Makefile:56]: Can we get rid of this SUFFIX stuff as well?

Comment (alecholmez) [reply Makefile:56]: Sure thing, I left that in there so I didn't have to touch the scripts but i'll rip it out

Comment (alecholmez) [reply Makefile:56]: @snowp do you want me to rip out the v2 bootstrap configs as well?

Comment (snowp) [reply Makefile:56]: Sure if it's no longer being used for anything

**Comment by alecholmez** [comment]
Should be good to go now @snowp @mattklein123

**Comment by mattklein123** [comment]
Hmm, I have no idea how I closed this: confused. And I don't see how to reopen it. Can you make a new PR?

**Comment by mattklein123** [comment]
Oh I guess this was based on my PR

**Comment by alecholmez** [comment]
@mattklein123 oops yes it was haha, I can make it against main

**Pull request closed without merging**